### DISCUSSION TITLE
Can more character types be supported? Such as wchar_t, char8_t, char16_t, char32_t

### DISCUSSION THREAD

**Discussion by aSurgingRiver** [nlohmann/json#4461] [General]
The software library supports the char character type by default.
Using wchar_t character type instantiation will result in an error.

basic_json<std::map, std::vector, std::wstring>

The library uses a large number of char instantiations in serialization and deserialization. This causes a mismatch with the wchar_t type.

**Comment by nlohmann** [comment]
The library's internal string storage is UTF-8 based, so `StringType` needs a `char`-compatible value type; `std::wstring`/`char16_t`/`char32_t`-based strings can't be used directly as `StringType`. You can still parse/store wide-character data by converting to/from UTF-8 first — see the FAQ's 'Wide string handling' section (https://json.nlohmann.me/home/faq/#wide-string-handling) for a `std::wstring` example using `codecvt_utf8`. `char8_t` is supported for the `_json`/`_json_pointer` literals as of recent versions.

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*
